### PR TITLE
feat(tpu): add create/delete/get samples 

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description
Sample in Python [feat: TPU *Create / *get / *create_with_script / *delete Samples by Thoughtseize1 · Pull Request #12690 · GoogleCloudPlatform/python-docs-samples](https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12690)
Documentation - https://cloud.google.com/tpu/docs/managing-tpus-tpu-vm
Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [x] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: tpu/src/test/java/tpu/TpuVmIT.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+@RunWith(JUnit4.class)
+@Timeout(value = 25, unit = TimeUnit.MINUTES)
+@TestMethodOrder(MethodOrderer. OrderAnnotation. class)
+public class TpuVmIT {
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String ZONE = "europe-west4-b";
+  static String javaVersion = System.getProperty("java.version").substring(0, 2);
+  private static final String TPU_VM_NAME = "test-tpu-" + javaVersion + "-"
+      + UUID.randomUUID().toString().substring(0, 8);
+  private static final String ACCELERATOR_TYPE = "v5litepod-1";
```
Comment (m-strzelczyk) [inline tpu/src/test/java/tpu/TpuVmIT.java:61]: Do we have access to v5lite? Can we rather use some older and cheaper TPU model for testing?

File: tpu/src/test/java/tpu/TpuVmIT.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+  public static void cleanUpExistingTpu(String prefixToDelete, String projectId, String zone)
+      throws IOException, ExecutionException, InterruptedException {
+    try (TpuClient tpuClient = TpuClient.create()) {
+      String parent = String.format("projects/%s/locations/%s", projectId, zone);
+      for (Node node : tpuClient.listNodes(parent).iterateAll()) {
+        String creationTime = formatTimestamp(node.getCreateTime());
+        String name = node.getName().substring(node.getName().lastIndexOf("/") + 1);
+        if (containPrefixToDeleteAndZone(node, prefixToDelete, zone)
+            && isCreatedBeforeThresholdTime(creationTime)) {
```
Comment (m-strzelczyk) [inline tpu/src/test/java/tpu/TpuVmIT.java:152]: If the test case takes < 5 minutes, do we just not delete the VM we created? Can we guarantee there won't be a TPU VM left running after the test runs once?

File: tpu/src/test/java/tpu/TpuVmIT.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+
+    // Cleanup existing stale resources.
+    cleanUpExistingTpu("test-tpu-" + javaVersion, PROJECT_ID, ZONE);
+  }
+
+  @AfterAll
+  public static void cleanup() throws Exception {
+    DeleteTpuVm.deleteTpuVm(PROJECT_ID, ZONE, TPU_VM_NAME);
+    TimeUnit.MINUTES.sleep(5);
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply tpu/src/test/java/tpu/TpuVmIT.java:84]: It depends :) This is long term operation. I've got error and after set timeout it disappeared.

File: tpu/src/test/java/tpu/TpuVmIT.java (hunk 2 above)
Comment (TetyanaYahodska) [reply tpu/src/test/java/tpu/TpuVmIT.java:152]: Thank you! Fixed time to 30 minutes. After implementing all samples for TPU we can set it to 24 hours for example.

File: tpu/src/test/java/tpu/TpuVmIT.java (hunk 1 above)
Comment (TetyanaYahodska) [reply tpu/src/test/java/tpu/TpuVmIT.java:61]: Changed accelerator type and version.

File: tpu/src/test/java/tpu/TpuVmIT.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply tpu/src/test/java/tpu/TpuVmIT.java:84]: Implemented OperationTimedPollAlgorithm with RetrySettings for TpuClient which permits to call delete method and don't use any timeouts.

**Comment by Sita04** [comment]
@TetyanaYahodska Please ping for review post @m-strzelczyk's LGTM.

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**